File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/Array_List/PreSum/minSubArrayLen_209.java
package algo.tzashinorpu.SpecialSubject.Array_List.PreSum;

public class minSubArrayLen_209 {
    public int minSubArrayLen(int target, int[] nums) {
        // target = 7, nums = [2,3,1,1,2,4,3]
        int len = nums.length, ans = len + 1;
        for (int i = 0, j = 0; j < len; j++) {
            target -= nums[j];
            while (target <= 0) {
                ans = Math.min(ans, j - i + 1);
                target += nums[i++];
            }
        }
        return ans % (len + 1);
    }
}

File: Clion/zero/1400_1600/minSubArrayLen_209.cpp
//
// Created by tseringnorpu on 6/14/2023.
//
#include "../header/header.h"
using namespace zero;

int Solution::minSubArrayLen(int target, vector<int> &nums) {
    int n = nums.size(),i=0,j=0;
    int res=n+1;
    for (; j < n; j++) {
        target -= nums[j];
        while (target <= 0) {
            res = min(res, j - i + 1);
            target += nums[i++];
        }
    }
    return res%(n+1);
}